src/uri.ts

```diff
@@ -34,27 +31,39 @@ export function resolveURI(uri: string): ResolvedURI {
  */
 export function codecovParamsForRepositoryCommit(
     uri: Pick<ResolvedURI, 'repo' | 'rev'>
-): Pick<CodecovGetCommitCoverageArgs, 'service' | 'owner' | 'repo' | 'sha'> {
-    // TODO: Support services (code hosts) other than GitHub.com, such as GitHub Enterprise, GitLab, etc.
-    if (uri.repo.startsWith('github.com/')) {
-        const parts = uri.repo.split('/', 4)
-        if (parts.length !== 3) {
-            throw new Error(
-                `invalid GitHub.com repository: ${JSON.stringify(
-                    uri.repo
-                )} (expected "github.com/owner/repo")`
-            )
-        }
+): Pick<CodecovGetCommitCoverageArgs, 'baseURL' | 'service' | 'owner' | 'repo' | 'sha'> {
+    try {
+        const endpoints: Endpoint[] | undefined = sourcegraph.configuration.get<Settings>().get('codecov.endpoints')
+        const baseURL: string = endpoints && endpoints[0] && endpoints[0].url || ''
+
+        const knownHosts: { name: string, service: string }[] = [
+            { name: 'github.com', service: 'gh' },
+            { name: 'gitlab.com', service: 'gl' },
+            { name: 'bitbucket.org', service: 'bb' },
+        ];
+
+
+        const knownHost: any = knownHosts.find(knownHost => uri.repo.includes(knownHost.name))
```

Comment:
```suggestion
        const knownHost = knownHosts.find(knownHost => uri.repo.includes(knownHost.name))
```

This type should be inferred, which is more safe than `any`